src/NATS.Client.Core/Internal/SslClientAuthenticationOptionsExtensions.cs

```diff
@@ -18,7 +18,41 @@ public static SslClientAuthenticationOptions LoadClientCertFromPem(this SslClien
             intermediateCerts.RemoveAt(0);
         }
 
-#if NET6_0
+#if !NET8_0_OR_GREATER
+        if (intermediateCerts.Count > 0)
+        {
+            throw new NotSupportedException("Client Certificates with intermediates are only supported in net8.0 and higher");
+        }
+#endif
+
+        // On Windows, ephemeral keys/certificates do not work with schannel. e.g. unless stored in certificate store.
+        // https://github.com/dotnet/runtime/issues/66283#issuecomment-1061014225
+        // https://github.com/dotnet/runtime/blob/380a4723ea98067c28d54f30e1a652483a6a257a/src/libraries/System.Net.Security/tests/FunctionalTests/TestHelper.cs#L192-L197
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            var ephemeral = leafCert;
+            leafCert = new X509Certificate2(leafCert.Export(X509ContentType.Pfx));
+            ephemeral.Dispose();
+        }
```

Comment:
why is this ephemeral trick moved up here?

Reply:
The ephemeral thing didn’t work with a PFX loaded with default options, it threw an exception that it couldn’t be exported